File: BOJ/15904.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
  vector<char> characters = {'U', 'C', 'P', 'C'};
  string s;
  getline(cin, s);

  for (auto c : characters) {
    auto n = s.find(c);
    if (n == string::npos) {
      cout << "I hate UCPC" << endl;
      return 0;
    }
    s = s.substr(n + 1);
  }
  cout << "I love UCPC" << endl;
  return 0;
}

File: BOJ/15904.cs
var inputLine = Console.ReadLine();



string Process(string sentence)
{
  var characters = new[]{
        'U', 'C', 'P', 'C'
    };
  foreach (var c in characters)
  {
    var n = sentence.IndexOf(c);
    if (n == -1)
    {
      return "I hate UCPC";
    }

    sentence = sentence[(n + 1)..];
  }

  return "I love UCPC";
}

Console.WriteLine(Process(inputLine));